csp/tests/test_decorators.py

```diff
@@ -19,94 +23,314 @@ def view(request):
+
+    response = view_without_decorator(REQUEST)
+    mw.process_response(REQUEST, response)
+    assert response[HEADER] == "default-src 'self'"
+    assert response[REPORT_ONLY_HEADER] == "default-src 'self'"
+
+
 @override_settings(CSP_IMG_SRC=['foo.com'])
 def test_csp_update():
     def view_without_decorator(request):
         return HttpResponse()
     response = view_without_decorator(REQUEST)
     mw.process_response(REQUEST, response)
-    policy_list = sorted(response['Content-Security-Policy'].split("; "))
+    policy_list = sorted(response[HEADER].split("; "))
     assert policy_list == ["default-src 'self'", "img-src foo.com"]
 
     @csp_update(IMG_SRC='bar.com')
     def view_with_decorator(request):
         return HttpResponse()
     response = view_with_decorator(REQUEST)
-    assert response._csp_update == {'img-src': 'bar.com'}
+    assert dict(response._csp_update) == {'default': {'img-src': ['bar.com']}}
     mw.process_response(REQUEST, response)
-    policy_list = sorted(response['Content-Security-Policy'].split("; "))
+    policy_list = sorted(response[HEADER].split("; "))
     assert policy_list == ["default-src 'self'", "img-src foo.com bar.com"]
 
     response = view_without_decorator(REQUEST)
     mw.process_response(REQUEST, response)
-    policy_list = sorted(response['Content-Security-Policy'].split("; "))
+    policy_list = sorted(response[HEADER].split("; "))
     assert policy_list == ["default-src 'self'", "img-src foo.com"]
 
 
+@override_settings(CSP_POLICIES=('default', 'report'))
+def test_csp_update_multiple():
+    @csp_update(
+        default={'img-src': 'bar.com'},
+        report={'font-src': 'foo.com'},
+    )
+    def view_with_decorator(request):
+        return HttpResponse()
+    response = view_with_decorator(REQUEST)
+    assert response._csp_update == {
+        'default': {'img-src': 'bar.com'},
+        'report': {'font-src': 'foo.com'},
+    }
+    mw.process_response(REQUEST, response)
+    policy_list = sorted(response[HEADER].split("; "))
+    assert policy_list == ["default-src 'self'", "img-src bar.com"]
+    policy_list = sorted(response[REPORT_ONLY_HEADER].split("; "))
+    assert policy_list == ["default-src 'self'", "font-src foo.com"]
+
+
 @override_settings(CSP_IMG_SRC=['foo.com'])
 def test_csp_replace():
     def view_without_decorator(request):
         return HttpResponse()
     response = view_without_decorator(REQUEST)
     mw.process_response(REQUEST, response)
-    policy_list = sorted(response['Content-Security-Policy'].split("; "))
+    policy_list = sorted(response[HEADER].split("; "))
     assert policy_list == ["default-src 'self'", "img-src foo.com"]
 
     @csp_replace(IMG_SRC='bar.com')
     def view_with_decorator(request):
         return HttpResponse()
     response = view_with_decorator(REQUEST)
-    assert response._csp_replace == {'img-src': 'bar.com'}
+    assert dict(response._csp_replace) == {'default': {'img-src': ['bar.com']}}
```

Comment:
Remove spurious `dict` coersion in tests (leftover from OrderedDict implementation)